lib/internal/http.js

```diff
@@ -132,6 +132,13 @@ class ProxyConfig {
     };
   }
 
+  // Private helper method for performance optimization
+  #normalizeHostname(hostname) {
+    return hostname.startsWith('[') && hostname.endsWith(']') ?
```

Comment:
Can you check the performance if you do not use startsWith and endsWith?

Usually direct access is faster...

```suggestion
    return hostname[0] === '[' && hostname[hostname.length - 1] === ']' ?
```

Reply:
This was already in benchmark file in comparison which resulted _**normalize: 205-337M ops/sec**_ which when tested was slower than **stringOp: ~267M-354M ops/sec**.  

```js
function normalizeProxyHostname(hostname) {
  if (hostname.length > 2 && 
      hostname[0] === '[' && 
      hostname[hostname.length - 1] === ']') {
    return hostname.slice(1, -1);
  }
  return hostname;
}
```

Reply:
I checked it further... it is impossible that the hostname starts with a '[' and does not end with ']'. The URLParser will throw an Error. So you just need to check if it is starting with '['.

Reply:
```suggestion
    return hostname[0] === '[' ?
```

Reply:
The issue is more, that you dont have a hotpath. This is -as far as i can see -only called once at startup. I am not even sure, if it makes sense to have a separate function, if it is only used in ProxyConfig.